### PR TITLE
Add public function to depsolv from a pool

### PR DESCRIPTION
**Pull request by r4f4**
This can be useful for applications that do not work with repos in the
filesystem, e.g, with databases. As long as the pool is pre-populated
with repos, the depsolving still works.

Signed-off-by: Rafael Fonseca <r4f4rfs@gmail.com>

### CODE REVIEW

File: fus.c [hunk 1]
```diff
@@ -1037,3 +1037,79 @@ fus_depsolve (const char *arch,
...
+ * returns: a NULL-terminated array of solved dependencies
+ */
+GPtrArray *
+fus_depsolve_from_pool (Pool       *pool,
+                        GHashTable *lookaside_repos,
+                        const GStrv exclude_packages,
+                        const GStrv solvables,
+                        GError    **error)
+{
+  g_assert (pool != NULL);
```
Comment (igor-raits) [inline fus.c:1059]: g_return_val_if_fail, asserts are getting optimized.

**Comment by igor-raits** [comment]
Do you have some use-case in mind for this?

File: fus.c (hunk 1 above)
Comment (r4f4) [reply fus.c:1059]: Oh ok.

**Comment by r4f4** [comment]
Yes, for Pulp like we were discussing on IRC the other day. I'll ask someone from Pulp to chime in.

**Comment by igor-raits** [comment]
the problem here is that you won't be able to pass libsolv's Python pool to here...

**Comment by r4f4** [comment]
Even if we write a SWIG python binding to fus?

**Comment by igor-raits** [comment]
since we are using GObject here, SWIG/Cython is no-go. We need to use GObject Introspection. But you would need to write gobject introspected bindings for libsolv... And I don't think anyone wants to do that.

**Comment by r4f4** [comment]
So do you think the easiest way would be to re-implement the algorithm in python?

**Comment by dparalen** [comment]
@r4f4 @ignatenkobrain you guys are 1h ahead of my current headache :D
@r4f4 thanks for putting this up
@ignatenkobrain so if we used some list of list of dicts as a poor-mans pool, could that work?
 
PS sorry for stupid questions, I don't know GObject/SWIG/Cython

**Comment by dparalen** [comment]
> @r4f4 @ignatenkobrain you guys are 1h ahead of my current headache :D
> @r4f4 thanks for putting this up
> @ignatenkobrain so if we used some list of list of dicts as a poor-mans pool, could that work?
> 
> PS sorry for stupid questions, I don't know GObject/SWIG/Cython

Answering myself: this would mean basically wrapping a lot of `libsolv` functionality in Fus just to be able to expose the API thru `pygobject`. Sounds like a terrible idea.

**Comment by dralley** [comment]
What about making it accessible via a native (C) Python extension?  

(FWIW I don't have much experience with this either)

**Comment by igor-raits** [comment]
What exactly you want to make accessible?

**Pull request closed without merging**